File: test/Stunts.Tests/StuntDiscovererTests.Compile.vb
Imports System
Imports Stunts

Public Class Tests
    Public Sub WhenFakingFormatterThenCanInvokeIt()
        Dim stunt1 As ICustomFormatter = Stunt.Of(Of ICustomFormatter)
        Dim stunt2 As ICustomFormatter = Stunt.Of(Of ICustomFormatter)

        Dim result1 As String = stunt1.Format("Hello {0}", "World", Nothing)
        Dim result2 As String = stunt2.Format("Hello {0}", "World", Nothing)
    End Sub
End Class

File: src/Stunts/Stunts.Tests/StuntDiscovererTests.Compile.cs
using System;
using Stunts;

public class Tests
{
    public void WhenFakingFormatterThenCanInvokeIt()
    {
        var stunt1 = Stunt.Of<ICustomFormatter>();
        var stunt2 = Stunt.Of<ICustomFormatter>();

        var result1 = stunt1.Format("Hello {0}", "World", null);
        var result2 = stunt2.Format("Hello {0}", "World", null);
    }
}